uefi-raw/src/protocol/network/snp.rs

```diff
@@ -0,0 +1,407 @@
+        self.to_option(self.tx_undersize_frames)
+    }
+
+    /// The number of frames longer than the maximum length for
+    /// the media. This would be 1500 for Ethernet
+    #[must_use]
+    pub const fn tx_oversize_frames(&self) -> Option<u64> {
+        self.to_option(self.tx_oversize_frames)
+    }
+
+    /// The number of valid frames that were dropped because
+    /// received buffers were full
+    #[must_use]
+    pub const fn tx_dropped_frames(&self) -> Option<u64> {
+        self.to_option(self.tx_dropped_frames)
+    }
+
+    /// The number of valid unicast frames transmitted and not
+    /// dropped
+    #[must_use]
+    pub const fn tx_unicast_frames(&self) -> Option<u64> {
+        self.to_option(self.tx_unicast_frames)
+    }
+
+    /// The number of valid broadcast frames transmitted and
+    /// not dropped
+    #[must_use]
+    pub const fn tx_broadcast_frames(&self) -> Option<u64> {
+        self.to_option(self.tx_broadcast_frames)
+    }
+
+    /// The number of valid multicast frames transmitted
+    /// and not dropped
+    #[must_use]
+    pub const fn tx_multicast_frames(&self) -> Option<u64> {
+        self.to_option(self.tx_multicast_frames)
+    }
+
+    /// The number of transmitted frames with CRC or
+    /// alignment errors
+    #[must_use]
+    pub const fn tx_crc_error_frames(&self) -> Option<u64> {
+        self.to_option(self.tx_crc_error_frames)
+    }
+
+    /// The total number of bytes transmitted including
+    /// error frames and dropped frames
+    #[must_use]
+    pub const fn tx_total_bytes(&self) -> Option<u64> {
+        self.to_option(self.tx_total_bytes)
+    }
+
+    /// The number of collisions detected on this subnet
+    #[must_use]
+    pub const fn collisions(&self) -> Option<u64> {
+        self.to_option(self.collisions)
+    }
+
+    /// The number of frames destined for unsupported protocol
+    #[must_use]
+    pub const fn unsupported_protocol(&self) -> Option<u64> {
+        self.to_option(self.unsupported_protocol)
+    }
+
+    /// The number of valid frames received that were duplicated
+    #[must_use]
+    pub const fn rx_duplicated_frames(&self) -> Option<u64> {
+        self.to_option(self.rx_duplicated_frames)
+    }
+
+    /// The number of encrypted frames received that failed
+    /// to decrypt
+    #[must_use]
+    pub const fn rx_decrypt_error_frames(&self) -> Option<u64> {
+        self.to_option(self.rx_decrypt_error_frames)
+    }
+
+    /// The number of frames that failed to transmit after
+    /// exceeding the retry limit
+    #[must_use]
+    pub const fn tx_error_frames(&self) -> Option<u64> {
+        self.to_option(self.tx_error_frames)
+    }
+
+    /// The number of frames that transmitted successfully
+    /// after more than one attempt
+    #[must_use]
+    pub const fn tx_retry_frames(&self) -> Option<u64> {
+        self.to_option(self.tx_retry_frames)
+    }
+}
+
+/// The Simple Network Mode
```

Comment:
nit: I think this description adds little value

```suggestion
/// Information about the current mode of a interface obtained by the Simple Network Protocol.
```